File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/VolumeMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Volume
import com.phamtunglam.health_connector_hc_android.pigeon.VolumeDto

/**
 * Converts a [VolumeDto] to a Health Connect [Volume] object.
 */
internal fun VolumeDto.toHealthConnect(): Volume = Volume.liters(liters)

/**
 * Converts a Health Connect [Volume] object to a [VolumeDto].
 *
 * Uses liters as the transfer unit for consistency.
 */
internal fun Volume.toDto(): VolumeDto = VolumeDto(
    liters = inLiters,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/volume_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Volume;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show VolumeDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Volume] to [VolumeDto].
@internal
extension VolumeToDto on Volume {
  VolumeDto toDto() {
    // Uses liters as the transfer unit for consistency.
    return VolumeDto(liters: inLiters);
  }
}

/// Converts [VolumeDto] to [Volume].
@internal
extension VolumeDtoToDomain on VolumeDto {
  Volume toDomain() {
    return Volume.liters(liters);
  }
}
